Project/Fall2020_CSC403_Project/FrmInventory.cs

```diff
@@ -0,0 +1,167 @@
+            {
+                instance = new FrmInventory(inventory,player);
+                instance.Instance();
+                
+            }
+            else if (!instance.IsActive())
+            {
+                instance = new FrmInventory(inventory, player);
+                instance.Instance();
+            }
+            instance.Show();
+        }
+
+        // Returns if the inventory updater thread is active.
+        public bool IsActive()
+        {
+            return active;
+        }
+
+        // Forces the thread updating the inventory to abort when the
+        // inventory window is closed.
+        private void FrmIventory_FormClosing(object sender, EventArgs e)
+        {
+            active = false;
+            invUpdater.Abort();
+            
+        }
+
+        // Initializes the inventory UI.
+        private void InitInvSlots()
+        {
+            inventorySlots = new Button[inventory.GetMaxInventorySize()];
+            for (int i = 1; i <= inventory.GetMaxInventorySize(); i++)
+            {
+                inventorySlots[i - 1] = Controls.Find("InvSlot" + i.ToString(), true)[0] as Button;
+                inventorySlots[i - 1].Tag = i-1;
+            }
+        }
+        
+        // Handles clicks on the inventory's buttons.
+        private void InvSlot_Click(object sender, EventArgs e)
+        {
+            Button invSlot = sender as Button;
+            int index = (int)invSlot.Tag;
+            inventory.UseInventoryObject(index, player);
+        }
+
+        // The UI is updated every 250ms, meant to be used in a separate thread
+        // from the main game.
+        private void ContinuousUpdater()
+        {
+            while (active)
+            {
+                UpdateInventory();
+                Thread.Sleep(50);
+            }
+            
+        }
+
+        // Updates the inventory ui to match what is stored in the inventory.
+        // This method is meant to be called from a secondary thread.
+        private void UpdateInventory()
+        {
+            for (int i = 0; i < inventory.GetMaxInventorySize(); i++)
+            {
+                if (inventory.GetInventoryObject(i) != null)
+                {
+                    InventoryObject iObj = inventory.GetInventoryObject(i);
+                    inventorySlots[i].Image = iObj.img;
+                    if (iObj.IsStackable())
+                    {
+
+                        // Allows the inventory updater thread to update the UI elements
+                        // in the inventory from outside of the main thread.
+                        try
+                        {
+                            this.Invoke(new MethodInvoker(delegate ()
+                            {
+                                inventorySlots[i].Text = inventory.GetInventoryObject(i).GetCount().ToString();
+                            }));
+                        }
+                        catch(Exception e)
```

Comment:
Should we be doing anything when we catch all these exceptions? If we're not specifying the exception on any of these or doing anything once it's caught, why not wrap the whole thing in a single try catch?